v3/src/components/graph/components/scatterdots.tsx

```diff
@@ -164,6 +175,125 @@ export const ScatterDots = observer(function ScatterDots(props: PlotProps) {
+      // attributes or the presence of a parent attribute and the number of unique values for that attribute. If there
+      // are multiple Y attributes, the number of groups matches the number of Y attributes. Otherwise, if there's a
+      // parent attribute, the number of groups matches the number of unique values for that attribute. If there's only
+      // one Y attribute and no parent attribute, then there's a only single group. The code below builds lists of
+      // connecting lines and case IDs for each group.
+      const lineGroups: Record<string, IConnectingLineDescription[]> = {}
+      const allLineCaseIds: Record<string, string[]> = {}
+      const yAttrCount = dataConfiguration?.yAttributeIDs?.length ?? 0
+      connectingLines.forEach((lineDescription: IConnectingLineDescription) => {
+        const parentAttrValue = parentAttrID ? importValueToString(lineDescription.caseData[parentAttrID]) : undefined
+        const groupKey = yAttrCount > 1
+          ? lineDescription.plotNum
+          : parentAttrValues && parentAttrValue
+            ? parentAttrValues.indexOf(parentAttrValue)
+            : 0
+
+        if (dataConfiguration?.isCaseInSubPlot(cellKey, lineDescription.caseData)) {
+          lineGroups[groupKey] ||= []
+          allLineCaseIds[groupKey] ||= []
+          lineGroups[groupKey].push(lineDescription)
+          allLineCaseIds[groupKey].push(lineDescription.caseData.__id__)
+        }
+      })
+
+      // For each group of lines, draw a path using the lines' coordinates
+      for (const [linesIndex, [primaryAttrValue, cases]] of Object.entries(lineGroups).entries()) {
+        const allLineCoords = cases.map((l) => l.lineCoords)
+        const lineCaseIds = allLineCaseIds[linesIndex]
+        const allCasesSelected = lineCaseIds?.every(caseID => dataConfiguration?.selection.includes(caseID))
+        const legendID = dataConfiguration?.attributeID("legend")
+        const color = parentAttrID && legendID
+          ? graphModel.pointDescription.pointColorAtIndex(linesIndex)
+          : graphModel.pointDescription.pointColorAtIndex(0)
+
+        connectingLinesArea
+          .append("path")
+          .data([allLineCoords])
+          .attr("d", d => curve(d))
+          .classed("selected", allCasesSelected)
+          .on("click", (e) => handleConnectingLinesClick(e, lineCaseIds))
+          .on("mouseover", (e) => handleConnectingLinesHover(e, lineCaseIds, parentAttrName, primaryAttrValue))
+          .on("mouseout", dataTip.hide)
+          .call(dataTip)
+          .attr("fill", "none")
+          .attr("stroke", color)
+          .attr("stroke-width", allCasesSelected ? 4 : 2)
+          .style("cursor", "pointer")
+          .style("opacity", connectingLinesActivatedRef.current ? 1 : 0)
+          .transition()
+          .duration(1000)
+          .style("opacity", showConnectingLines ? 1 : 0)
+          .on("end", connectingLinesCleanUp)
```

Comment:
As I just found out recently, the `end` function is called once for each element being iterated which was a problem in a map with a bunch of points, but may not be as big a deal with a presumably small number of lines.